src/main/java/org/jvnet/solaris/libzfs/ZFSObject.java

```diff
@@ -486,21 +523,30 @@ public void setProperty(String key, String value) {
      */
     public Set<ZFSSnapshot> snapshots() {
         final Set<ZFSSnapshot> set = new TreeSet<ZFSSnapshot>();
-        String abi = library.getFeature("LIBZFS4J_ABI_zfs_iter_snapshots");
+        String abi_thisfunc = "snapshots";
+        String abi_toggle = "LIBZFS4J_ABI_zfs_iter_snapshots";
+        String abi = library.getFeature(abi_toggle);
+        if (abi.equals("NO-OP")) {
+            LOGGER.log(Level.FINE, "NO-OP: libzfs4j::" + abi_thisfunc + "() was called while " + abi_toggle + "=='" + abi + "' - skipped due to config");
+        } else
         if (abi.equals("openzfs")) {
             LIBZFS.zfs_iter_snapshots(handle, false, new libzfs.zfs_iter_f() {
                 public int callback(zfs_handle_t handle, Pointer arg) {
                     set.add((ZFSSnapshot)ZFSObject.create(library, handle));
                     return 0;
                 }
             }, null);
-        } else {
+        } else
+        if (abi.equals("legacy")) {
```

Comment:
Unfortunately, with the way this evolution went, there is nothing like numeric comparison similar to `v1 < v1 < v3` that would work for the whole set of differing features.

The initial idea per discussions with @kohsuke and @xenol was sort of naïve, using a big toggle for "legacy" or "new" ABI - but in fact every routine can have some variant in a certain OS distribution version, as their maintainers pick these or those OpenZFS patches from upstream, or develop their own.

So the initial implementation revolves around notions of "legacy" and "openzfs" settings, which works for most of the routines, where "legacy" is a common ground for these distros shared mostly by latest relatively freely licensed sol10u8, and opensolaris. Then they parted ways.

In this vein, things like pre-nvXX or pre-solXX are older than legacy and only handled because research into ABI evolution showed these changes even exist ;)